homeassistant/components/solaredge/sensor.py

```diff
@@ -159,9 +293,137 @@ def update(self):
+
+    @Throttle(INVENTORY_UPDATE_DELAY)
+    def update(self):
+        """Update the data from the SolarEdge Monitoring API."""
+        from requests.exceptions import HTTPError, ConnectTimeout
+
+        try:
+            data = self.api.get_inventory(self.site_id)
+            inventory = data['Inventory']
+        except KeyError:
+            _LOGGER.error("Missing inventory data, skipping update")
+            return
+        except (ConnectTimeout, HTTPError):
+            _LOGGER.error("Could not retrieve data, skipping update")
+            return
+
+        self.data = {}
+        self.attributes = {}
+
+        for key, value in inventory.items():
+            self.data[key] = len(value)
+            self.attributes[key] = {key: value}
+
+        _LOGGER.debug("Updated SolarEdge inventory: %s, %s",
+                      self.data, self.attributes)
+
+
+class SolarEdgePowerFlowDataService(SolarEdgeDataService):
+    """Get and update the latest power flow data."""
+
+    def __init__(self, api, site_id):
+        """Initialize the power flow data service."""
+        super().__init__(api, site_id)
+
+        self.unit = ''
+
+    @Throttle(POWER_FLOW_UPDATE_DELAY)
+    def update(self):
+        """Update the data from the SolarEdge Monitoring API."""
+        from requests.exceptions import HTTPError, ConnectTimeout
+
+        try:
+            data = self.api.get_current_power_flow(self.site_id)
+            power_flow = data['siteCurrentPowerFlow']
+        except KeyError:
+            _LOGGER.error("Missing power flow data, skipping update")
+            return
+        except (ConnectTimeout, HTTPError):
+            _LOGGER.error("Could not retrieve data, skipping update")
+            return
+
+        power_from = []
+        power_to = []
+
+        if 'connections' not in power_flow:
+            _LOGGER.error("Missing connections in power flow data")
+            return
+
+        for connection in power_flow['connections']:
+            power_from.append(connection['from'].lower())
+            power_to.append(connection['to'].lower())
+
+        self.data = {}
+        self.attributes = {}
+        self.unit = power_flow['unit']
+
+        for key, value in power_flow.items():
+            if key in ['LOAD', 'PV', 'GRID', 'STORAGE']:
+                self.data[key] = value['currentPower']
+                self.attributes[key] = {'status': value['status']}
+
+            if key in ['GRID']:
+                export = key.lower() in power_to
+                self.data[key] *= -1 if export else 1
+                self.attributes[key]['flow'] = ('export' if export
+                                                else 'import')
+
+            if key in ['STORAGE']:
+                charge = key.lower() in power_to
+                self.data[key] *= -1 if charge else 1
+                self.attributes[key]['flow'] = ('charge' if charge
+                                                else 'discharge')
+                pass
```

Comment:
Why `pass`?

Reply:
@MartinHjelmare Thanks for the review! You're right, will remove this.